docs/sql_reference/src/asciidoc/_chapters/sql_utilities.adoc

```diff
@@ -48,6 +49,326 @@ to connect to and issue SQL utilities, see the
+* `CLEANUP UID {object-uid}`
++
+This command will remove all entries from all metadata tables that match the specified object UID. If the specified object UID exists in the OBJECTS metadata table and the object name associated with that UID can be retrieved, then that object is also removed from HBase.
+
+* `CLEANUP METADATA [ , CHECK ] [ , RETURN DETAILS ]`
++
+This command will look for inconsistent entries in the metadata, return information about them, and clean them up. Multiple checks are done to validate metadata consistency.
+
+** `CHECK`
++
+One can check for inconsistent metadata entries without actually cleaning them up.
+This can be done by specifying the `CHECK` option. If `CHECK` is specified, the following checks will be performed simultaneously, and a summary of inconsistent objects belonging to the categories listed below will be returned.
+
+*** Orphan Metadata Objects
++
+This check is done to validate that objects that exist in metadata also exist in HBase. If the object does not exist in HBase, then it is marked as an orphan metadata object.
+
+*** Orphan HBase Objects
++
+This check is done to validate that objects that exist in HBase also exist in metadata. If the object does not exist in metadata, then it is marked as an orphan HBase object.
+
+*** Inconsistent Object Entries
++
+This check is done to make sure that all entries in the OBJECTS table also exist in
+the OBJECTS unique index (OBJECTS_UNIQ_IDX) and that all entries in OBJECTS_UNIQ_IDX also exist in the OBJECTS table. It also checks to make sure that all entries in the COLUMNS table have a corresponding entry in the OBJECTS table. If any of the previous checks fail, then that entry is marked as inconsistent.
+
+*** Inconsistent View Entries
++
+This check is done to make sure that all view entries are consistent among the views and the OBJECTS table.
+
+** `RETURN DETAILS`
++
+If `RETURN DETAILS` is specified, then details on which objects are inconsistent is also returned.
+
+<<<
+[[cleanup_considerations]]
+=== Considerations for CLEANUP Statement
+
+[[security_and_privileges]]
+==== Security and Privileges
+
+To clean up an object, one needs to have DROP privileges. If object privileges cannot be retrieved from the metadata or if metadata cleanup is being done, then one needs to be DB_ROOT to perform the cleanup.
+
+[[consistency_checks_during_access]]
+==== Consistency Checks During Access
+
+If an object exists in the metadata but does not exist in HBase, then error 4254 will be returned if that object is accessed during any DML or DDL operation.
+
+An object is also marked as inconsistent during any complex DDL operation. For example, if a CREATE TABLE also creates implicit indexes or constraints, then the object will be marked as inconsistent until the complex DDL operation is completed. If an abnormal error happens (like a process crash or a node failure), then that object remains inconsistent. Any attempt to access that object will return error 4254.

```

Comment:
"(like a process crash...)"  Wordsmith nit. "(such as a process crash...)" is a bit better because here you are citing process crashes and node failures as examples of errors, rather than characterizing all errors to be similar to process crashes or node failures, which is what "like" implies.